modules/transcription-service-google-speech-impl/src/main/java/org/opencastproject/transcription/googlespeech/GoogleSpeechTranscriptionService.java

```diff
@@ -0,0 +1,1064 @@
+        }
+        logger.info("Environment name is {}", clusterName);
+
+        logger.info("Activated!");
+      } else {
+        logger.info("Service disabled. If you want to enable it, please update the service configuration.");
+      }
+    } else {
+      throw new IllegalArgumentException("Missing component context");
+    }
+  }
+
+  @Override
+  public Job startTranscription(String mpId, Track track, String language) throws TranscriptionServiceException {
+    if (!enabled) {
+      throw new TranscriptionServiceException(
+              "This service is disabled. If you want to enable it, please update the service configuration.");
+    }
+
+    try {
+      return serviceRegistry.createJob(JOB_TYPE, Operation.StartTranscription.name(),
+              Arrays.asList(mpId, MediaPackageElementParser.getAsXml(track), language));
+    } catch (ServiceRegistryException e) {
+      throw new TranscriptionServiceException("Unable to create a job", e);
+    } catch (MediaPackageException e) {
+      throw new TranscriptionServiceException("Invalid track " + track.toString(), e);
+    }
+  }
+
+  @Override
+  public Job startTranscription(String mpId, Track track) throws TranscriptionServiceException {
+    throw new UnsupportedOperationException("Not supported.");
+  }
+
+  @Override
+  public void transcriptionDone(String mpId, Object obj) throws TranscriptionServiceException {
+    JSONObject jsonObj = null;
+    String jobId = null;
+    String token = INVALID_TOKEN;
+    try {
+      token = getRefreshAccessToken();
+    } catch (IOException ex) {
+      logger.error("Unable to create access token, error: {}", ex.toString());
+    }
+    if (token.equals(INVALID_TOKEN)) {
+      throw new TranscriptionServiceException("Invalid access token");
+    }
+    try {
+      jsonObj = (JSONObject) obj;
+      jobId = (String) jsonObj.get("name");
+      logger.info("Transcription done for mpId {}, jobId {}", mpId, jobId);
+      JSONArray resultsArray = getTranscriptionResult(jsonObj);
+
+      // Update state in database
+      // If there's an optimistic lock exception here, it's ok because the workflow dispatcher
+      // may be doing the same thing
+      database.updateJobControl(jobId, TranscriptionJobControl.Status.TranscriptionComplete.name());
+
+      // Delete audio file from Google storage
+      deleteStorageFile(mpId, token);
+
+      // Save results in file system if there exist
+      if (resultsArray != null) {
+        saveResults(jobId, jsonObj);
+      }
+    } catch (IOException e) {
+      logger.warn("Could not save transcription results file for mpId {}, jobId {}: {}",
+              new Object[]{mpId, jobId, jsonObj == null ? "null" : jsonObj.toJSONString()});
+      throw new TranscriptionServiceException("Could not save transcription results file", e);
+    } catch (TranscriptionDatabaseException e) {
+      logger.warn("Transcription results file were saved but state in db not updated for mpId {}, jobId {}", mpId,
```

Comment:
yes line 378, it could be a race condition but main motivation is for example if Google storage services or its API's are inaccessible at the time of deletion, job should not be mark as completed to avoid orphan file on google storage.

Reply:
Ah, good point.  Ok, let's leave this then.